### ISSUE TITLE
failing process58 test

### ISSUE THREAD

**Issue by ybznek** [OpenSCAP/openscap#493]
```
+ local PID=
++ local 'item=/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]'
++ assert_exists 1 '/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]'
++ sed 's|&gt;|>|g'
++ sed 's|&lt;|<|g'
+++ xpath_wrapper command_line.out.pkh7zK 'count(/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""])'
+++ '[' 2 == 1 ']'
+++ '[' 2 == 2 ']'
+++ file=command_line.out.pkh7zK
+++ xpath_expr='count(/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""])'
+++ /bin/xpath -e 'count(/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""])' command_line.out.pkh7zK
Query didn't return a nodeset. Value: ++ real_cnt=0
++ '[' 0 '!=' 1 ']'
++ echo 'Failed: expected count: 1, real count: 0, xpath: '\''/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]'\'''
++ return 1
++ xpath_wrapper command_line.out.pkh7zK '/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]/../unix-sys:command_line/text()'
+ cmdline='Failed: expected count: 1, real count: 0, xpath: '\''/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]'\'''
+ echo 'Failed: expected count: 1, real count: 0, xpath: '\''/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]'\'''
+ grep -qE '^\[command_line.sh\] <defunct>$'
+ echo 'Failed: expected command-line of process with PID= is '\''^\[command_line.sh\] <defunct>$'\'' but real is: '\''Failed: expected count: 1, real count: 0, xpath: '\''/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]'\'''\'''
Failed: expected command-line of process with PID= is '^\[command_line.sh\] <defunct>$' but real is: 'Failed: expected count: 1, real count: 0, xpath: '/oval_results/results/system/oval_system_characteristics/system_data/unix-sys:process58_item[@status="exists"]/unix-sys:pid[text()=""]''
```

I don't think that it is related to our memory issues.

**Comment by jan-cerny** [comment]
Yes, it should not been related to the memory issue. We have seen this same log already multiple times. My opinion is that there is a race condition. Some process got terminated before we could get its PID.

Could we set running the background processes for a longer time? Tests on Jenkins nodes are sometimes very slow, especially with valgrind enabled, so the process could timed out before it was examined by oscap.

**Comment by mpreisler** [comment]
Looks like a race condition.

**Comment by evgenyz** [comment]
Do we still have this problem in 1.3? Are we going to fix it in 1.2?
